File: Tyche.Android/app/src/main/java/com/pipel/tyche/poolLayout/domain/PoolLayout.kt
package com.pipel.tyche.poolLayout.domain

import com.pipel.core.type.DateTime
import com.pipel.core.type.NonEmptyString
import com.pipel.core.type.Ulid

data class PoolLayout(
    val poolLayoutId: Ulid,
    val name: NonEmptyString,
    val openingStartDateTime: DateTime,
    val openingEndDateTime: DateTime
)

File: Tyche.Backend/Pipel.Tyche.PoolLayout/Pipel.Tyche.PoolLayout/Domain/PoolLayout.fs
namespace Pipel.Tyche.PoolLayout.Domain

open Pipel.Type

type PoolLayoutPK =
    { PoolLayoutId: Ulid }

type PoolLayout =
    { PoolLayoutPK: PoolLayoutPK
      Name: NonEmptyString100
      OpeningStartDateTime: DateTime
      OpeningEndDateTime: DateTime }
